src/GslibIO.jl

```diff
@@ -43,6 +51,38 @@ function load(file::File{format"GSLIB"})
     georef(data, domain)
   end
 end
+"""
+    parse_gslib(filename::AbstractString)
+
+Helper function to parse a standard GSLIB file.
+It returns the data as a matrix `X`, and `metadata` that includes the parsed information.
+"""
+
+function parse_gslib(filename::AbstractString)
+  open(filename) do fs
+    # header
+    header = readline(fs)
+    # number of variables is the first integer in the second line (it may contain extra info)
+    line = readline(fs)
+    linesplit = split(line)
+    nvars = parse(Int, linesplit[1])
+    if length(linesplit) > 1
+      loc = findfirst(linesplit[1], line)
+      extrainfo = lstrip(line[loc.stop+1:end]) # extrainfo is kept as it is after `nvars`
+    else
+      extrainfo = "" # empty in case extrainfo is not provided
+    end
+    # properties names
+    vars = [strip(readline(fs)) for i in 1:nvars]
+
+    metadata = GSLIBMetadata(header, nvars, vars, extrainfo)
+
+    # read data
+    data = readdlm(fs)
+
+    data, metadata
```

Comment:
So you read the spec and the data table, but then return in the opposite order. Please call the variables `meta` and `data` and return `meta, data`.